src/execution_plan/optimizations/optimizations_util.c

```diff
@@ -0,0 +1,59 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "optimizations_util.h"
+
+void re_order_filter_op(ExecutionPlan *plan, OpBase *root, OpFilter *filter) {
+	FT_FilterNode *additional_filter_tree = filter->filterTree;
+
+	rax *references = FilterTree_CollectModified(additional_filter_tree);
+	OpBase *op = ExecutionPlan_LocateReferences(root, NULL, references);
+	ExecutionPlan_RemoveOp(plan, (OpBase *)filter);
+	ExecutionPlan_PushBelow(op, (OpBase *)filter);
+	raxFree(references);
+}
+
+// Returns true if the stream resolves all required entities.
+static bool _stream_resolves_entities(rax *stream_resolves, rax *entities_to_resolve) {
```

Comment:
I think this function is equivalent to `raxIsSubset` in utils/rax_extensions.h.

Reply:
yep. nice!
changed